test-integration/server/statistics/KeyspaceStatisticsIT.java

```diff
@@ -0,0 +1,295 @@
+        TransactionOLTP tx = localSession.transaction().write();
+        AttributeType ageType = tx.putAttributeType("age", AttributeType.DataType.LONG);
+        Role friend = tx.putRole("friend");
+        EntityType personType = tx.putEntityType("person").plays(friend).has(ageType);
+        RelationType friendshipType = tx.putRelationType("friendship").relates(friend);
+        tx.commit();
+
+        tx = localSession.transaction().write();
+        ageType = tx.getAttributeType("age");
+        Attribute age = ageType.create(1);
+        personType = tx.getEntityType("person");
+        Entity person1 = personType.create().has(age).has(age);
+        Entity person2 = personType.create().has(age);
+        friendshipType = tx.getRelationType("friendship");
+        friend = tx.getRole("friend");
+        friendshipType.create().assign(friend, person1).assign(friend, person2);
+        tx.close();
+
+        tx = localSession.transaction().write();
+        long personCount = localSession.keyspaceStatistics().count(tx, "person");
+        long ageCount = localSession.keyspaceStatistics().count(tx, "age");
+        long friendshipCount = localSession.keyspaceStatistics().count(tx, "friendship");
+        long implicitAgeCount = localSession.keyspaceStatistics().count(tx, "@has-age");
+        tx.close();
+
+        assertEquals(0, personCount);
+        assertEquals(0, ageCount);
+        assertEquals(0, friendshipCount);
+        assertEquals(0, implicitAgeCount);
+    }
+
+    /**
+     * Tests persistence
+     */
+    @Test
+    public void reopeningSessionRetrievesStatistics() {
+        TransactionOLTP tx = localSession.transaction().write();
+        AttributeType ageType = tx.putAttributeType("age", AttributeType.DataType.LONG);
+        Role friend = tx.putRole("friend");
+        EntityType personType = tx.putEntityType("person").plays(friend).has(ageType);
+        RelationType friendshipType = tx.putRelationType("friendship").relates(friend);
+
+        ageType = tx.getAttributeType("age");
+        Attribute age = ageType.create(1);
+        personType = tx.getEntityType("person");
+        Entity person1 = personType.create().has(age).has(age);
+        Entity person2 = personType.create().has(age);
+        friendshipType = tx.getRelationType("friendship");
+        friend = tx.getRole("friend");
+        friendshipType.create().assign(friend, person1).assign(friend, person2);
+        tx.commit();
+
+        tx = localSession.transaction().write();
+        long personCount = localSession.keyspaceStatistics().count(tx, "person");
+        long ageCount = localSession.keyspaceStatistics().count(tx, "age");
+        long friendshipCount = localSession.keyspaceStatistics().count(tx, "friendship");
+        long implicitAgeCount = localSession.keyspaceStatistics().count(tx, "@has-age");
+        tx.close();
+
+        localSession.close();
+
+        // at this point, the graph and keyspace should be deleted from Grakn server
```

Comment:
just by closing the session graph and keyspace are deleted?

Reply:
yep!

Reply:
What happens is that there's an `onSessionClose` callback that deletes the Keyspace, graph, statistics etc. when the last session closes

Reply:
ok so we are referring to caches, not to graph/keyspace itself, ok